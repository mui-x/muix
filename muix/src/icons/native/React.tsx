import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)