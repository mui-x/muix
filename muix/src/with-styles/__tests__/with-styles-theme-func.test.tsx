/** @jsx platform.createElement */
import React from 'react'

import { TSheeter } from 'reactxx-typings'
import { platform } from 'reactxx-sheeter'

import { ts, traceComponentEx, Shape, ReactAny, theme, ThemeProvider } from "reactxx-tests"

describe("WITH STYLES THEME", () => {
  it("WEB", () => traceTest(true));
  it("NATIVE", () => traceTest(false));
});

const customTheme: TSheeter.getTheme<Shape> = {
  ...theme,
  secondary: {
    ...theme.secondary,
    normal: {
      ...theme.secondary.normal,
      background: 'lightcyan',
    },
  },
}

const traceTest = (isWeb: boolean) => {

  traceComponentEx(isWeb, { traceLevel: 2},

    theme => ({
      label: theme.secondary.normal,
      root: {},
      webOnly: {},
      nativeOnly: {},
    }),

    ({ classes, classNameX, toClassNames, styleX, children }) =>
      <ReactAny classNameX={toClassNames([classes.label, classNameX])} styleX={styleX}>{children}</ReactAny>,

    Comp => <ReactAny>
      <Comp
        classes={theme => ({ label: { color: theme.secondary.normal.background } })}
        classNameX={theme => ({ backgroundColor: theme.secondary.normal.background })}
        styleX={theme => ({ borderBottomColor: theme.secondary.normal.background })}
      >Default Theme</Comp>
      <ThemeProvider theme={customTheme}>
        <Comp
          classes={theme => ({ label: { color: theme.secondary.normal.background } })}
          classNameX={theme => ({ backgroundColor: theme.secondary.normal.background })}
          styleX={theme => ({ borderBottomColor: theme.secondary.normal.background })}
        >Custom Theme</Comp>
      </ThemeProvider>
    </ReactAny>,

    {
      defaultProps: {
        classes: theme => ({ root: { borderLeftColor: theme.secondary.normal.background } }),
        classNameX: theme => ({ borderRightColor: theme.secondary.normal.background }),
        styleX: theme => ({ borderTopColor: theme.secondary.normal.background }),
      }
    }
  )
}
