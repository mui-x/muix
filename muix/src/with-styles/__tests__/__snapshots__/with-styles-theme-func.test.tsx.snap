// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WITH STYLES THEME NATIVE 1`] = `
<ReactAny>
  <ReactAny
    style={
      Object {
        "background": "lightblue",
        "backgroundColor": "lightblue",
        "borderBottomColor": "lightblue",
        "borderRightColor": "lightblue",
        "borderTopColor": "lightblue",
        "color": "lightblue",
      }
    }
    trace={
      Object {
        "background": "lightblue /*TestComponent.sheet.label*/",
        "backgroundColor": "lightblue /*classNameX*/",
        "borderBottomColor": "lightblue /*styleX*/",
        "borderRightColor": "lightblue /*TestComponent.option.classNameX*/",
        "borderTopColor": "lightblue /*TestComponent.option.styleX*/",
        "color": "lightblue /*classes.label*/",
      }
    }
  >
    Default Theme
  </ReactAny>
  <ThemeProviderGeneric>
    <ReactAny
      style={
        Object {
          "background": "lightcyan",
          "backgroundColor": "lightcyan",
          "borderBottomColor": "lightcyan",
          "borderRightColor": "lightcyan",
          "borderTopColor": "lightcyan",
          "color": "lightcyan",
        }
      }
      trace={
        Object {
          "background": "lightcyan /*TestComponent.sheet.label*/",
          "backgroundColor": "lightcyan /*classNameX*/",
          "borderBottomColor": "lightcyan /*styleX*/",
          "borderRightColor": "lightcyan /*TestComponent.option.classNameX*/",
          "borderTopColor": "lightcyan /*TestComponent.option.styleX*/",
          "color": "lightcyan /*classes.label*/",
        }
      }
    >
      Custom Theme
    </ReactAny>
  </ThemeProviderGeneric>
</ReactAny>
`;

exports[`WITH STYLES THEME WEB 1`] = `
<ReactAny>
  <ReactAny
    className="f d e b"
    style={
      Object {
        "borderBottomColor": "lightblue",
        "borderTopColor": "lightblue",
      }
    }
    trace={
      Array [
        ".f { background-color:lightblue /*classNameX*/ }",
        ".d { border-right-color:lightblue /*TestComponent.option.classNameX*/ }",
        ".e { color:lightblue /*classes.label*/ }",
        ".b { background:lightblue /*TestComponent.sheet.label*/ }",
      ]
    }
  >
    Default Theme
  </ReactAny>
  <ThemeProviderGeneric>
    <ReactAny
      className="k i j g"
      style={
        Object {
          "borderBottomColor": "lightcyan",
          "borderTopColor": "lightcyan",
        }
      }
      trace={
        Array [
          ".k { background-color:lightcyan /*classNameX*/ }",
          ".i { border-right-color:lightcyan /*TestComponent.option.classNameX*/ }",
          ".j { color:lightcyan /*classes.label*/ }",
          ".g { background:lightcyan /*TestComponent.sheet.label*/ }",
        ]
      }
    >
      Custom Theme
    </ReactAny>
  </ThemeProviderGeneric>
</ReactAny>
`;
